src/server/tests.rs

```diff
@@ -1437,6 +1507,84 @@ x + 1
     );
 }
 
+#[test]
+async fn test_hover_with_markdown() {
+    let fluxscript = r#"x = 1
+x + 1
+"#;
+    let server = create_server();
+    let params = lsp::InitializeParams {
+        capabilities: lsp::ClientCapabilities {
+            workspace: None,
+            text_document: Some(
+                lsp::TextDocumentClientCapabilities {
+                    synchronization: None,
+                    completion: None,
+                    hover: Some(lsp::HoverClientCapabilities {
+                        dynamic_registration: Some(true),
+                        content_format: Some(vec![
+                            lsp::MarkupKind::Markdown,
+                        ]),
+                    }),
+                    signature_help: None,
+                    references: None,
+                    document_highlight: None,
+                    document_symbol: None,
+                    formatting: None,
+                    range_formatting: None,
+                    on_type_formatting: None,
+                    declaration: None,
+                    definition: None,
+                    type_definition: None,
+                    implementation: None,
+                    code_action: None,
+                    code_lens: None,
+                    document_link: None,
+                    color_provider: None,
+                    rename: None,
+                    publish_diagnostics: None,
+                    folding_range: None,
+                    selection_range: None,
+                    linked_editing_range: None,
+                    call_hierarchy: None,
+                    semantic_tokens: None,
+                    moniker: None,
```

Comment:
In this situation, where you're taking `None` for all the values you don't want to specify, you can do something like this:

```
lsp::TextDocumentClientCapabilities {
    hover: Some(lsp::HoverClientCapabilities {
        dynamic_registration: Some(true),
        content_form: Some(vec![
            lsp::MarkupKind::Markdown,
        ]),
    },
    ..Default::default()
}
```

The by-product of this is that you can see exactly what's expected from the client for the test.